### PR TITLE
Fetch users

### PR DESCRIPTION
**Pull request by alisa-lin**
## Change Log
- Share clink menu now populates with users, and you can search and select users
## Screenshots
Before
Screenshot 2020-07-22 at 12 21 42 PM
After
Screenshot 2020-07-22 at 12 22 12 PM
Screenshot 2020-07-22 at 12 22 37 PM
## Next Steps
Write to database upon submission so as to actually share to these users.

### CODE REVIEW

**Comment by michaeleggboy** [comment]
remove package files

File: react-app/src/features/users.js [hunk 1]
```diff
@@ -113,4 +127,15 @@ export const checkUser = () => async dispatch => {
   }
 }
 
+export const fetchUsers = (token) => async dispatch => {
+  try {
+    dispatch(fetchUsersStart())
+    const response = await apis.fetchUsers(token)
+    console.log(response);
```
Comment (marsw42) [inline react-app/src/features/users.js:134]: Remove console.log

File: react-app/src/components/Common/clinkmenu.js [hunk 2]
```diff
@@ -11,135 +13,145 @@ const layout = {
...
-    });
+  const showShare = () => {
+    setShareVisible(true);
   };
 
-  onEditFinish = values => {
-    console.log('Success:', values);
-    this.setState({ loading: true });
+  const onEditFinish = values => {
+    setLoading(true);
```
Comment (marsw42) [inline react-app/src/components/Common/clinkmenu.js:41]: Are these just mocks?

**Comment by michaeleggboy** [comment]
Don't merge in yet

File: react-app/src/components/Common/clinkmenu.js (hunk 2 above)
Comment (alisa-lin) [reply react-app/src/components/Common/clinkmenu.js:41]: yeah actually, we've been doing all of them like that and i forgot to make them real LOL

Comment (alisa-lin) [reply react-app/src/components/Common/clinkmenu.js:41]: Yes it's just a mock, the actual dispatch is coming in a separate PR!

File: react-app/src/components/Common/topbar.js [hunk 3]
```diff
@@ -15,7 +15,7 @@ const Topbar = () => {
 
   const currentToken = localStorage.getItem('currentToken')
   const currentUser = useSelector(state => state.users.currentUser)
-  const title = useSelector(state => state.clink.currentClink);
+  const title = useSelector(state => state.clink.currentClink)
```
Comment (marsw42) [inline react-app/src/components/Common/topbar.js:18]: Add semicolon back

File: flask-server/main.py [hunk 4]
```diff
@@ -60,22 +60,26 @@ def store_user():
...
-        array.append([user['email'], user['username'], user['password_hash'], user['registered_on'], user['deleted']])
-    return jsonify(array)
+def fetch_users():
+    resp_token = decode_auth_token(request.headers.get('Authorization'))
+    if is_valid_instance(resp_token):
+        query = datastore_client.query(kind='user').add_filter('deleted', '=', False)
+        users = list(query.fetch())
+        array = []
+        for user in users:
+            if user.id != str(resp_token):
```
Comment (michaeleggboy) [inline flask-server/main.py:71]: remove the str(), the id is an integer. the if statement will always be true and you won't remove the current user.

**Pull request merged**